kuzzleio/kuzzle-device-manager
# feat(engine): add mappings update conflict ask call

## What does this PR do ?
This PR add an ask call to check for any conflict with new mappings to be detected and returned, allowing the current update operation to be aborted without performing any changes. This check is for now applied on the API and database level of models updates.


### How should this be manually tested?


  - Step 1 : Add a field to the mappings associated to any type of models via the API or directly from the console.
  - Step 2 : Change this field type and receive an error from the back end.
  - Step 3 : Check the model type collection mappings to see if the changes were persisted.

### Boyscout


Added more documentation to the DeviceManagerEngine file

Added documentation web pages about potential API errors
![image](https://github.com/kuzzleio/kuzzle-device-manager/assets/167076031/cfa737b9-6276-4c6a-9398-89fe908f1f17)
